pkg/sink/sink.go

```diff
@@ -0,0 +1,131 @@
+	"github.com/tidwall/gjson"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeclientset "k8s.io/client-go/kubernetes"
+)
+
+type Resource struct {
+	TriggersClient         *triggersclientset.Clientset
+	KubeClient             *kubeclientset.Clientset
+	EventListenerName      string
+	EventListenerNamespace string
+}
+
+func (r Resource) HandleEvent(response http.ResponseWriter, request *http.Request) {
+	el, err := r.TriggersClient.TektonV1alpha1().EventListeners(r.EventListenerNamespace).Get(r.EventListenerName, metav1.GetOptions{})
+	if err != nil {
+		log.Printf("Error getting EventListener %s in Namespace %s: %s", r.EventListenerName, r.EventListenerNamespace, err)
+		return
+	}
+
+	event, err := ioutil.ReadAll(request.Body)
+	if err != nil {
+		log.Printf("Error reading event body: %s", err)
+		return
+	}
+	log.Printf("Event: %s", string(event))
+
+	for _, tbRef := range el.Spec.TriggerBindingRefs {
+		tbNamespace := tbRef.Namespace
+		if tbNamespace == "" {
+			tbNamespace = r.EventListenerNamespace
+		}
+		tb, err := r.TriggersClient.TektonV1alpha1().TriggerBindings(tbNamespace).Get(tbRef.Name, metav1.GetOptions{})
+		if err != nil {
+			log.Printf("Error getting TriggerBinding %s in Namespace %s: %s", tbRef.Name, tbNamespace, err)
+			continue
+		}
+
+		for _, templateBinding := range tb.Spec.TemplateBindings {
+			ttNamespace := templateBinding.TemplateRef.Namespace
+			if ttNamespace == "" {
+				ttNamespace = r.EventListenerNamespace
+			}
+			tt, err := r.TriggersClient.TektonV1alpha1().TriggerTemplates(ttNamespace).Get(templateBinding.TemplateRef.Name, metav1.GetOptions{})
+			if err != nil {
+				log.Printf("Error getting TriggerTemplate %s in Namespace %s: %s", templateBinding.TemplateRef.Name, ttNamespace, err)
+				continue
+			}
+
+			params, err := template.ApplyEventToParams(event, templateBinding.Params)
+			if err != nil {
+				log.Printf("Error applying event to TriggerBinding params: %s", err)
+				continue
+			}
+			params = template.AddDefaultParamsFromSpecs(params, tt.Spec.Params)
+
+			for _, rt := range tt.Spec.ResourceTemplates {
+				rtRaw := template.ApplyParamsToResourceTemplate(params, rt.RawMessage)
+				err = createResource(r.KubeClient, rtRaw)
+				if err != nil {
+					log.Printf("Error creating resources from resource template %s: %s", string(rtRaw), err)
+					continue
+				}
+				log.Printf("Successfully created resource: %s", string(rt.RawMessage))
+			}
+		}
+	}
+}
+
+// createResource uses the kubeClient to create the resource defined in the
+// TriggerResourceTemplate and returns any errors with this process
+func createResource(kubeClient *kubeclientset.Clientset, rt json.RawMessage) error {
+	// Assume the TriggerResourceTemplate is valid (it has an apiVersion and Kind)
+	result := kubeClient.RESTClient().Post().
+		RequestURI(getRequestURI(rt)).
+		Body([]byte(rt)).
+		SetHeader("Content-Type", "application/json").
+		Do()
+	if result.Error() != nil {
```

Comment:
I'm think about if exception occurred when creating resource what should end user do?
- firstly, we should let user know error happen (update status of `eventListener`?)
- then, let user correct it then `apply eventListener` again?
if so, i think we need add something like:
`if apierrors.IsAlreadyExists(result.Error()) {` 
since we don't clean old resource when exception raised last time.

BTW, could we add `owner` for these resource?

Reply:
> I'm think about if exception occurred when creating resource what should end user do?

This is a good point about error handling. In the current implementation, an error would only be displayed in the pod logs. Since we haven't discussed this type of error handling in too much depth, I didn't want to implement anything too complicated in this PR. I like your idea of surfacing errors in the EventListener's status.

I think we should open a new issue to discuss and implement this error handling; should I open a new issue, or do you want to handle this @vincent-pli? 👍 

> then, let user correct it then apply eventListener again?
if so, i think we need add something like:
if apierrors.IsAlreadyExists(result.Error()) {
since we don't clean old resource when exception raised last time.

This is an interesting idea about "re-applying" failed resources. I'm not sure if it will be useful, but I think this indicates that we should have some way of keeping a record of the Resource with its injected event data. At the moment, I do [log the JSON of the resource](https://github.com/tektoncd/triggers/pull/74/files#diff-57fed74cee793226b40869dac1113d50R89) if it fails to be created. I'm not sure if there's anything else I can do in this PR to improve this issue.

> BTW, could we add owner for these resource?

I chose to leave the `owner` alone for the moment, because I'd like to tackle that in the [existing house keeper issue](https://github.com/tektoncd/triggers/issues/64). Plus, I think that once we're able to use the Triggers project, we will gain many insights into how we actually want to clean up the generated resources.

Reply:
I guess we need consider these issues, seems they are all corner cases but will sharply improve usability, I will log some issue to trace them, thanks

Reply:
Thanks!